src/code/z_skin_matrix.c

```diff
@@ -1,42 +1,692 @@
+        mf_copy.mf[3][this_row] = mf_copy.mf[3][this_row] / diag_element;
+        dest->mf[3][this_row] = dest->mf[3][this_row] / diag_element;
+
+        // col i = col i - a * col j
+        for (k = 0; k != 4; k++) { //
+            if (k != this_row) {
+                temp2 = mf_copy.mf[this_row][k];
+                mf_copy.mf[0][k] = mf_copy.mf[0][k] - (mf_copy.mf[0][this_row] * temp2);
+                dest->mf[0][k] = dest->mf[0][k] - (dest->mf[0][this_row] * temp2);
+                mf_copy.mf[1][k] = mf_copy.mf[1][k] - (mf_copy.mf[1][this_row] * temp2);
+                dest->mf[1][k] = dest->mf[1][k] - (dest->mf[1][this_row] * temp2);
+                mf_copy.mf[2][k] = mf_copy.mf[2][k] - (mf_copy.mf[2][this_row] * temp2);
+                dest->mf[2][k] = dest->mf[2][k] - (dest->mf[2][this_row] * temp2);
+                mf_copy.mf[3][k] = mf_copy.mf[3][k] - (mf_copy.mf[3][this_row] * temp2);
+                dest->mf[3][k] = dest->mf[3][k] - (dest->mf[3][this_row] * temp2);
+            }
+        }
+    }
+    return 0;
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_matrix/func_800A73E0.s")
+#endif
+
+// Output matrix scales x,y,z components of vector or x,y,z rows of matrix when applied on LHS
+void SkinMatrix_Scale(MtxF* mf, f32 xScale, f32 yScale, f32 zScale) {
+    mf->xy = 0.0f;
+    mf->xz = 0.0f;
+    mf->xw = 0.0f;
+    mf->yx = 0.0f;
+    mf->yz = 0.0f;
+    mf->yw = 0.0f;
+    mf->zx = 0.0f;
+    mf->zy = 0.0f;
+    mf->zw = 0.0f;
+    mf->wx = 0.0f;
+    mf->wy = 0.0f;
+    mf->wz = 0.0f;
+    mf->ww = 1.0f;
+    mf->xx = xScale;
+    mf->yy = yScale;
+    mf->zz = zScale;
+}
+
+// output is (roll rotation matrix) * (pitch rotation matrix) * (yaw rotation matrix)
+void SkinMatrix_RotateRPY(MtxF* mf, s16 roll, s16 pitch, s16 yaw) {
+    f32 cos2;
+    f32 sin;
+    f32 cos;
+    f32 yx;
+    f32 sin2;
+    f32 zx;
+    f32 yy;
+    f32 zy;
+
+    sin = Math_Sins(yaw);
+    cos = Math_Coss(yaw);
+    mf->yy = cos;
+    mf->yx = -sin;
+    mf->xw = mf->yw = mf->zw = 0;
+    mf->wx = mf->wy = mf->wz = 0;
+    mf->ww = 1;
+
+    if (pitch != 0) {
+        sin2 = Math_Sins(pitch);
+        cos2 = Math_Coss(pitch);
+
+        mf->xx = cos * cos2;
+        mf->zx = cos * sin2;
+
+        mf->xy = sin * cos2;
+        mf->zy = sin * sin2;
+        mf->xz = -sin2;
+        mf->zz = cos2;
+
+    } else {
+        mf->xx = cos;
+        if (1) {} // required to match
+        if (1) {} // required to match
+        zx = sin; // required to match
+        mf->xy = sin;
+        mf->xz = mf->zx = mf->zy = 0;
+        mf->zz = 1;
+    }
+
+    if (roll != 0) {
+        sin2 = Math_Sins(roll);
+        cos2 = Math_Coss(roll);
+
+        yx = mf->yx;
+        zx = mf->zx;
+        mf->yx = (yx * cos2) + (zx * sin2);
+        mf->zx = (zx * cos2) - (yx * sin2);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_matrix/func_800A76A4.s")
+        if (1) {} // required to match
```

Comment:
we will assume that anything dumb like this will be required for matching, i dont think we should point it out with comments. same for all the other ones below